File: Leetcode/剑指offer/easy/20210913/deleteNode.cpp
class Solution {
public:
    ListNode* deleteNode(ListNode* head, int val) {
        ListNode *pre=new ListNode(-1);
        pre->next=head;
        while(pre->next->val!=val){
            pre=pre->next;
        }
        if(pre->next==head){
            return pre->next->next;
        }else{
            pre->next=pre->next->next;
        }
        return head;
    }
};

File: Leetcode/Review/20211004/deleteNode.go
package _0211004

func deleteNode(head *ListNode, val int) *ListNode {
	dummyHead:=&ListNode{Next:head}
	cur:=dummyHead
	for cur.Next.Val!=val{
		cur=cur.Next
	}
	cur.Next=cur.Next.Next
	return dummyHead.Next
}
